pkg/providers/nutanix/template.go

```diff
@@ -209,11 +211,26 @@ func buildTemplateMapMD(clusterSpec *cluster.Spec, workerNodeGroupMachineSpec v1
 	return values
 }
 
-func buildTemplateMapSecret(clusterSpec *cluster.Spec, creds []byte) map[string]interface{} {
+func buildTemplateMapSecret(secretName string, creds credentials.BasicAuthCredential) (map[string]interface{}, error) {
+	encodedCreds, err := jsonMarshal(creds)
+	if err != nil {
+		return nil, err
+	}
+
+	nutanixCreds := []credentials.Credential{{
+		Type: credentials.BasicAuthCredentialType,
+		Data: encodedCreds,
+	}}
+	credsJSON, err := jsonMarshal(nutanixCreds)
+	if err != nil {
+		return nil, err
+	}
+
 	values := map[string]interface{}{
-		"clusterName":              clusterSpec.Cluster.Name,
+		"secretName":               secretName,
```

Comment:
Since we are managing 2 secrets now, one that EKS-A consumes and the other that CAPX consumes, should we add a random string or hash at the end of `secretName` to ensure that the two secrets don't end up being named the same?

Reply:
Given the contents of the secrets are the same, if they do end up being the same, it's still not really an issue.

Reply:
Hmm, I see. Wouldn't adding a check to ensure they don't have the same name be a better strategy then rather than obfuscation by adding a random string?

Reply:
we could do that too but how do you determine that the secret you are referring to is being used by CAPX or EKS-A or something else? We would only want to prevent them from using the same secret as the one being used by CAPX.
If it's a secret being used by EKS-A, that's fair game.
If it's any other secret, then what do we do?

Reply:
EKS-A any value other than the cluster name and CAPX same value as cluster name as it's a cluster scoped secret?